pkg/engine/jmespath/functions.go

```diff
@@ -736,6 +747,23 @@ func jpLabelMatch(arguments []interface{}) (interface{}, error) {
 	return true, nil
 }
 
+func jpToBoolean(validateArg []interface{}) (interface{}, error) {
+	str, ok := validateArg[0].(string)
+	if !ok {
+		return nil, fmt.Errorf("jpToBoolean: argument must be a string")
+	}
+	lowerStr := strings.ToLower(str)
+
+	if lowerStr != "true" && lowerStr != "false" {
```

Comment:
ParseBool should throw an error when the input is invalid, we don't need an additional check here.

Reply:
@samj1912  Actually, we discussed in the last Kyverno meeting that this filter should not return true or false for "t", "1", "f", or "0". but if we remove this code it will not return error in this values.
LInk : https://www.golangprograms.com/string-to-boolean-data-type-conversion-in-go.html.

We are allowing only case sensitive like True, TRUe, FAlse, false etc.